src/main/java/amazin/service/SecurityServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package amazin.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SecurityServiceImpl implements SecurityService {
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    private static final Logger logger = LoggerFactory.getLogger(SecurityServiceImpl.class);
+
+    @Override
+    public String findLoggedInEmail() {
+        Object userDetails = SecurityContextHolder.getContext().getAuthentication().getDetails();
+        if (userDetails instanceof UserDetails) {
+            return ((UserDetails)userDetails).getUsername();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void autoLogin(String username, String password) {
+        UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+        UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken =
+	    new UsernamePasswordAuthenticationToken(userDetails, password,
+						    userDetails.getAuthorities());
+
+        authenticationManager.authenticate(usernamePasswordAuthenticationToken);
+
+        if (usernamePasswordAuthenticationToken.isAuthenticated()) {
+            SecurityContextHolder.getContext().setAuthentication(usernamePasswordAuthenticationToken);
+            logger.debug(String.format("Auto login %s successfully!", username));
```

Comment:
that's fine, the whole point of using a logger like this is so that you can run the application in debug mode to get more verbose output. This is pretty standard.

Reply:
doesn't really effect performance and is kind of helpful for debugging if we need it

Reply:
debugging a working component?

Reply:
> debugging a working component?
> wouldn't we end up with a log of users successfully logging in which clutter from important logs?

This is why it's a debug message, it means we want to know everything that's happening

Reply:
It is an important debug message as it logs the user in case of an issue caused, we would know the user affected. The level is configured to debug meaning only when verbosity is changed to debug we will see it.